src/ngraph/runtime/gpu/gpu_emitter.cpp

```diff
@@ -171,6 +171,247 @@ cudnnSetOpTensorDescriptor(opTensorDesc,
                 writer.block_end();
             }
 
+            template <>
+            void GPU_Emitter::EMITTER_DECL(ngraph::op::Convolution)
+            {
+                if (out[0].get_size() == 0)
+                {
+                    return;
+                }
+
+                const std::string args0 = "x_descriptor";
+                const std::string args1 = "w_descriptor";
+                const std::string out0 = "y_descriptor";
+                const std::string conv_descriptor = "conv_descriptor";
+                const std::string data_type = "CUDNN_DATA_FLOAT";
+                const std::string tensor_format = "CUDNN_TENSOR_NCHW";
+                const std::string mode = "CUDNN_CROSS_CORRELATION";
+                const std::string conv_algo = "CUDNN_CONVOLUTION_FWD_ALGO_IMPLICIT_GEMM";
+                const size_t workSapceSizeInBytes = 1024;
+                auto convolution = static_cast<const ngraph::op::Convolution*>(node);
+                Strides window_dilation_strides = convolution->get_window_dilation_strides();
+                Strides window_movement_strides = convolution->get_window_movement_strides();
+                Strides data_dilation_strides = convolution->get_data_dilation_strides();
+                CoordinateDiff padding = convolution->get_padding_below();
+                CoordinateDiff padding_above = convolution->get_padding_above();
+
+                if (padding.size() > 3)
```

Comment:
for convolution, it does not pad NC.

Reply:
Right, but pad = {0,0,1,1}, also does not pad NC, only HW. If the padding array is passed in as this, instead of {1,1}, it will it your assertion even though it is only 2 dimensional data.

Reply:
Maybe try this with one of the tests and see where it is caught?

Reply:
Looks like the vector is coming from nGraph, so it will not include the N, C dimensions.

Reply:
@aprocter In that case, I would suggest that the above mentioned test name is a bit misleading for op::Pad. In that test, it is an exterior pad of a 4d tensor, and the pad is provided as {0,0,1,1} which includes N and C. 

If I am supposed to interpret this not as a NCHW tensor, but rather just an NDArray, then it is only a partial exterior padding and then the name is wrong.

Please correct me if I am missing a third alternative interpretation, that's just where I got with this test :)

Reply:
will op::Pad actually pad NC?